go/pio/host/allwinner/allwinner.go

```diff
@@ -835,71 +862,224 @@ var mapping = [116][5]string{
+	{"SDC0_D0", "JTAG_DI1"},  // PF1
+	{"SDC0_CLK", "UART0_TX"}, // PF2
+	{"SDC0_CMD", "JTAG_DO1"}, // PF3
+	{"SDC0_D3", "UART0_RX"},  // PF4
+	{"SDC0_D2", "JTAG_CK1"},  // PF5
+	{}, // PF6
+	{"SDC1_CLK", "", "", "", "PG_EINT0"},             // PG0
+	{"SDC1_CMD", "", "", "", "PG_EINT1"},             // PG1
+	{"SDC1_D0", "", "", "", "PG_EINT2"},              // PG2
+	{"SDC1_D1", "", "", "", "PG_EINT3"},              // PG3
+	{"SDC1_D2", "", "", "", "PG_EINT4"},              // PG4
+	{"SDC1_D3", "", "", "", "PG_EINT5"},              // PG5
+	{"UART1_TX", "", "", "", "PG_EINT6"},             // PG6
+	{"UART1_RX", "", "", "", "PG_EINT7"},             // PG7
+	{"UART1_RTS", "", "", "", "PG_EINT8"},            // PG8
+	{"UART1_CTS", "", "", "", "PG_EINT9"},            // PG9
+	{"AIF3_SYNC", "PCM1_SYNC", "", "", "PG_EINT10"},  // PG10
+	{"AIF3_BCLK", "PCM1_BCLK", "", "", "PG_EINT11"},  // PG11
+	{"AIF3_DOUT", "PCM1_DOUT", "", "", "PG_EINT12"},  // PG12
+	{"AIF3_DIN", "PCM1_DIN", "", "", "PG_EINT13"},    // PG13
+	{"I2C0_SCL", "", "", "", "PH_EINT0"},             // PH0
+	{"I2C0_SDA", "", "", "", "PH_EINT1"},             // PH1
+	{"I2C1_SCL", "", "", "", "PH_EINT2"},             // PH2
+	{"I2C1_SDA", "", "", "", "PH_EINT3"},             // PH3
+	{"UART3_TX", "", "", "", "PH_EINT4"},             // PH4
+	{"UART3_RX", "", "", "", "PH_EINT5"},             // PH5
+	{"UART3_RTS", "", "", "", "PH_EINT6"},            // PH6
+	{"UART3_CTS", "", "", "", "PH_EINT7"},            // PH7
+	{"OWA_OUT", "", "", "", "PH_EINT8"},              // PH8
+	{"", "", "", "", "PH_EINT9"},                     // PH9
+	{"MIC_CLK", "", "", "", "PH_EINT10"},             // PH10
+	{"MIC_DATA", "", "", "", "PH_EINT11"},            // PH11
+	{"S_RSB_SCK", "S_I2C_SCL", "", "", "S_PL_EINT0"}, // PL0
+	{"S_RSB_SDA", "S_I2C_SDA", "", "", "S_PL_EINT1"}, // PL1
+	{"S_UART_TX", "", "", "", "S_PL_EINT2"},          // PL2
+	{"S_UART_RX", "", "", "", "S_PL_EINT3"},          // PL3
+	{"S_JTAG_MS", "", "", "", "S_PL_EINT4"},          // PL4
+	{"S_JTAG_CK", "", "", "", "S_PL_EINT5"},          // PL5
+	{"S_JTAG_DO", "", "", "", "S_PL_EINT6"},          // PL6
+	{"S_JTAG_DI", "", "", "", "S_PL_EINT7"},          // PL7
+	{"S_I2C_CSK", "", "", "", "S_PL_EINT8"},          // PL8
+	{"S_I2C_SDA", "", "", "", "S_PL_EINT9"},          // PL9
+	{"S_PWM", "", "", "", "S_PL_EINT10"},             // PL10
+	{"S_CIR_RX", "", "", "", "S_PL_EINT11"},          // PL11
+	{"", "", "", "", "S_PL_EINT12"},                  // PL12
+	{}, // PB10
+	{}, // PB11
+	{}, // PB12
+	{}, // PB13
+	{}, // PB14
+	{}, // PB15
+	{}, // PB16
+	{}, // PB17
+	{}, // PB18
+	{}, // PD25
+	{}, // PD26
+	{}, // PD27
+}
+
+// mappingR8 for Allwinner R8, excludes functions in and out.
+// Datasheet, page 18.
+// https://github.com/NextThingCo/CHIP-Hardware/raw/master/CHIP%5Bv1_0%5D/CHIPv1_0-BOM-Datasheets/Allwinner%20R8%20Datasheet%20V1.2.pdf
+//
+// - The datasheet uses TWI instead of I2C but I renamed for consistency.
```

Comment:
I wanted to comment this hadn't done. I also renamed things in bcm compared to spec so all functions are always consistent independent of the CPU dataset being used. This needs to be called out in the doc, was thinking about pins.ByFunction and actively enforce patterns when registering pins.